### PR TITLE
add relay quoter

### PR DESCRIPTION
**Pull request by snreynolds**

### CODE REVIEW

File: test/foundry-tests/RelayOrderQuoter.t.sol [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+            info: OrderInfo({reactor: reactor, swapper: swapper, nonce: 0, deadline: deadline}),
+            decayStartTime: block.timestamp,
+            decayEndTime: decayEndTime,
+            actions: actions,
+            inputs: inputs
+        });
+        bytes memory sig = signOrder(swapperPrivateKey, address(permit2), order);
+        vm.expectRevert(ReactorErrors.DeadlineBeforeEndTime.selector);
+        quoter.quote(abi.encode(order), sig);
+    }
```
Comment (zhongeric) [inline test/foundry-tests/RelayOrderQuoter.t.sol:102]: Would like to test nonce and sig related errors too, as well as errors bubbled up from the execution of the encoded actions

File: test/foundry-tests/RelayOrderQuoter.t.sol [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        assertEq(address(quoter.getReactor(abi.encode(order))), address(0xbeef));
+    }
+
+    function testQuote() public {
+        tokenIn.forceApprove(swapper, address(permit2), ONE);
+
+        Input[] memory inputs = new Input[](1);
+        bytes[] memory actions = new bytes[](0);
+
+        inputs[0] = Input({token: address(tokenIn), recipient: address(0), startAmount: ONE, maxAmount: ONE});
```
Comment (zhongeric) [inline test/foundry-tests/RelayOrderQuoter.t.sol:66]: Can we add an input here with address(0) recipient that we map to the caller's address and expect that it's returned in the resolved transfer details?

**Review by zhongeric — COMMENTED** [review]
looks good! I think its out of date from latest main but only a few comments

File: test/foundry-tests/RelayOrderQuoter.t.sol (hunk 2 above)
Comment (snreynolds) [reply test/foundry-tests/RelayOrderQuoter.t.sol:67]: https://github.com/Uniswap/relayer/pull/42#discussion_r1473396101 like here?

Comment (zhongeric) [reply test/foundry-tests/RelayOrderQuoter.t.sol:67]: oh nice yeah

File: test/foundry-tests/executors/RelayOrderExecutorTest.t.sol [hunk 3]
```diff
@@ -99,15 +99,15 @@ contract RelayOrderExecutorTest is Test, PermitSignature, DeployPermit2 {
     }
 
     // execute batch via multicall
-    function testExecutorMulticall() public {
-        address recipient = vm.addr(0x1);
-        tokenIn.mint(address(executor), ONE);
-        vm.deal(address(executor), ONE);
+    // function testExecutorMulticall() public {
```
Comment (snreynolds) [inline test/foundry-tests/executors/RelayOrderExecutorTest.t.sol:102]: Commented this out cause I was getting compiler warnings

**Review by zhongeric — APPROVED** [review]
looks great!

**Pull request merged**